### PR TITLE
[MPI/PETSc] Enable rank output.

### PR DESCRIPTION
**Pull request by TomFischer**

### CODE REVIEW

File: BaseLib/TemplateLogogFormatterSuppressedGCC-impl.h [hunk 1]
```diff
@@ -16,10 +16,7 @@ ::TemplateLogogFormatterSuppressedGCC(MPI_Comm mpi_comm)
 {
     int rank = 0;
     MPI_Comm_rank(mpi_comm, &rank);
-    int size = 0;
-    MPI_Comm_size(mpi_comm, &size);
-    int digits = size/10 + 1;
-    _str_mpi_rank = "rank " + padLeft(std::to_string(rank), digits) + "| ";
+    _str_mpi_rank = "[" + std::to_string(rank) + "] ";
```
Comment (chleh) [inline BaseLib/TemplateLogogFormatterSuppressedGCC-impl.h]: The former code didn't work?

File: Applications/CLI/ogs.cpp [hunk 2]
```diff
@@ -144,6 +145,11 @@ int main(int argc, char *argv[])
                 vtkSmartPointer<vtkMPIController>::New();
             controller->Initialize(&argc, &argv, 1);
             vtkMPIController::SetGlobalController(controller);
+
+            logog_setup.setFormatter(
+                std::make_unique<BaseLib::TemplateLogogFormatterSuppressedGCC<
+                    TOPIC_LEVEL_FLAG | TOPIC_FILE_NAME_FLAG |
+                    TOPIC_LINE_NUMBER_FLAG>>());
```
Comment (chleh) [inline Applications/CLI/ogs.cpp]: Will that automatically nicely format the `[mpi_rank]` line prefixes?

Comment (TomFischer) [reply Applications/CLI/ogs.cpp]: The output looks like:
```
[9] Assemble HTProcess.
[1] Assemble HTProcess.
[3] Assemble HTProcess.
[0] Assemble HTProcess.
```

File: BaseLib/TemplateLogogFormatterSuppressedGCC-impl.h (hunk 1 above)
Comment (TomFischer) [reply BaseLib/TemplateLogogFormatterSuppressedGCC-impl.h]: I think the padding was not computed correctly. Furthermore, my aim was to reduce the size of the message as much as possible but containing the relevant information.

**Pull request merged**